### ISSUE TITLE
Error when viewing via GameObject Explorer after Patch 1.

### ISSUE THREAD

**Issue by WitchGod** [ShinyHobo/BG3-Modders-Multitool#131]
After updating to Patch 1 and unpacking, here's the error when viewing via GameObject Explorer.

```
ading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Parallel.ForWorker[TLocal](Int32 fromInclusive, Int32 toExclusive, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, Action`1 body)
   at bg3_modders_multitool.Services.RootTemplateHelper.ReadVisualBanks()
   at bg3_modders_multitool.Services.RootTemplateHelper.<LoadRootTemplates>b__51_0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at bg3_modders_multitool.Services.RootTemplateHelper.<LoadRootTemplates>d__51.MoveNext()
[08/25/2023 21:39:03] Loading GameObjects...
[08/25/2023 21:39:26] GameObjects failed to load. 23.2565573 seconds passed.
[08/25/2023 21:39:26] One or more errors occurred.
[08/25/2023 21:39:26] One or more errors occurred.
[08/25/2023 21:39:26]    at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Parallel.ForWorker[TLocal](Int32 fromInclusive, Int32 toExclusive, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, Action`1 body)
   at bg3_modders_multitool.Services.RootTemplateHelper.ReadVisualBanks()
   at bg3_modders_multitool.Services.RootTemplateHelper.<LoadRootTemplates>b__51_0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at bg3_modders_multitool.Services.RootTemplateHelper.<LoadRootTemplates>d__51.MoveNext()
[08/25/2023 21:40:31] Unpacking processes starting. This could take a while; please wait for all console processes to close on their own.
[08/25/2023 21:40:33] All unpacking processes finished.
```

For reference, here's the items I've unpacked.

```
\ExportTool-v1.18.5\tools\UnpackedData\Assets
\ExportTool-v1.18.5\tools\UnpackedData\day0
\ExportTool-v1.18.5\tools\UnpackedData\Effects
\ExportTool-v1.18.5\tools\UnpackedData\English
\ExportTool-v1.18.5\tools\UnpackedData\Game
\ExportTool-v1.18.5\tools\UnpackedData\Gustav
\ExportTool-v1.18.5\tools\UnpackedData\Icons
\ExportTool-v1.18.5\tools\UnpackedData\Materials
\ExportTool-v1.18.5\tools\UnpackedData\Models
\ExportTool-v1.18.5\tools\UnpackedData\Patch0_Hotfix1
\ExportTool-v1.18.5\tools\UnpackedData\Patch0_Hotfix2
\ExportTool-v1.18.5\tools\UnpackedData\Patch0_Hotfix3
\ExportTool-v1.18.5\tools\UnpackedData\Patch0_Hotfix4
\ExportTool-v1.18.5\tools\UnpackedData\Patch1
\ExportTool-v1.18.5\tools\UnpackedData\Shared
\ExportTool-v1.18.5\tools\UnpackedData\Textures
```

Is this possibly an error with the new files or is this more of a performance issue?

**Comment by WitchGod** [comment]
@ShinyHobo Sorry about this but I reviewed the logs and further up, I saw this as well.

```
[08/25/2023 21:23:39] Retrieving file list for decompression.
[08/25/2023 21:23:43] Retrived file list. Starting decompression; this could take awhile.
[08/25/2023 21:23:43] Failed to convert resource: Invalid LSF signature; expected 464F534C, got 200A0D7B
[08/25/2023 21:23:43] Failed to convert resource: Invalid LSF signature; expected 464F534C, got 200A0D7B
[08/25/2023 21:23:52] Retrieving file list.
[08/25/2023 21:23:56] File list retrieved.
[08/25/2023 21:23:56] Starting index process.
[08/25/2023 21:36:51] Indexing process finished in 00:12:55.
```

I wasn't being too vigilant. Not too sure if this is tied to the original error. Thank you.

**Comment by ShinyHobo** [comment]
Multiple issues present with latest patch; sorting it out.

**Comment by Charizard748** [comment]
> @ShinyHobo Sorry about this but I reviewed the logs and further up, I saw this as well.
> 
> ```
> [08/25/2023 21:23:39] Retrieving file list for decompression.
> [08/25/2023 21:23:43] Retrived file list. Starting decompression; this could take awhile.
> [08/25/2023 21:23:43] Failed to convert resource: Invalid LSF signature; expected 464F534C, got 200A0D7B
> [08/25/2023 21:23:43] Failed to convert resource: Invalid LSF signature; expected 464F534C, got 200A0D7B
> [08/25/2023 21:23:52] Retrieving file list.
> [08/25/2023 21:23:56] File list retrieved.
> [08/25/2023 21:23:56] Starting index process.
> [08/25/2023 21:36:51] Indexing process finished in 00:12:55.
> ```
> 
> I wasn't being too vigilant. Not too sure if this is tied to the original error. Thank you.

I am also having this issue of 
> [08/25/2023 21:23:43] Failed to convert resource: Invalid LSF signature; expected 464F534C, got 200A0D7B

**Comment by ShinyHobo** [comment]
The "Invalid LSF signature" error is expected of files that are actually already lsx files that were improperly named by Larian. I am simply filtering them out in the next build.

**Comment by ShinyHobo** [comment]
This should be fixed in v0.9.5
